ts/src/main/scala/ch/epfl/ts/component/fetch/HistDataCSVFetcher.scala

```diff
@@ -0,0 +1,185 @@
+ * t is the original (historical) time difference between the quote that was last sent 
+ * and the next quote to be sent.
+ * 
+ * @param dataDir       A directory path containing data files the fetcher can read. Which files are
+ *                      actually read is determined by the start and end arguments (see below).
+ *                      The directory should contain substructures of the form <currency pair>/<xyz>.csv, e.g.: 
+ *                      EURCHF/DAT_NT_EURCHF_T_ASK_201304.csv,
+ *                      EURCHF/DAT_NT_EURCHF_T_BID_201304.csv,
+ *                      ...
+ *                      EURUSD/DAT_NT_EURUSD_T_BID_201305.csv, etc.
+ * @param currencyPair  The currency pair to be read from the data directory, e.g. "eurchf", "USDCHF", etc.
+ * @param start         From when to read. This can be any date, but will be reduced to its month. That means
+ *                      if start is set to 2013-04-24 14:34 the fetcher will start reading the first quote available
+ *                      in the data file for April 2013 (as if start was set to 2013-04-01 00:00).
+ * @param end           Until when to read. Behaves analogous to start, i.e. if end is set to 2013-06-24 14:34
+ *                      the fetcher will still read and send all data in June 2013, as if end was set to 2013-06-30 24:00 
+ * @param speed         The speed at which the fetcher will fetch quotes. Defaults to 1 (which means quotes are replayed
+ *                      as they were recorded). Can be set e.g. to 2 (time runs twice as fast) or 60 (one hour of historical
+ *                      quotes is sent in one minute), etc. Time steps are dT = int(1/speed * dT_historical), in milliseconds.
+ *                      Accordingly: be careful with high speeds, if e.g. speed=1000 the system can't make the difference
+ *                      between dT_historical=4200 and dT_historical=4900, they will be both sent after 4ms in the simulation.
+ *                      If we go even higher, dT will become < 1 and thus 0, which in theory would mean send all the quotes
+ *                      you read at the same time, but in practice would probably mess up things.
+ */
+
+class HistDataCSVFetcher(dataDir: String, currencyPair: String, 
+                         start: Date, end: Date,
+                         speed: Double = 1.0)  extends PushFetchComponent[Quote] {
+  
+  val workingDir = dataDir + "/" + currencyPair.toUpperCase() + "/";
+  val (whatC, withC) = Currency.pairFromString(currencyPair);
+  
+  val bidPref = "DAT_NT_" + currencyPair.toUpperCase() + "_T_BID_"
+  val askPref = "DAT_NT_" + currencyPair.toUpperCase() + "_T_ASK_"
+  
+  /**
+   * The centerpiece of this class, where we actually load the data.
+   * It contains all quotes this fetcher reads from disk, ready to be fetch()ed
+   */
+  val allQuotes: List[Quote] = monthsBetweenStartAndEnd
+                                .flatMap(m => parse(bidPref + m + ".csv", askPref + m + ".csv"))
```

Comment:
So wait, there *is* an expected format for the filenames. This should be documented above in the explanation about the `dataDir` param.